flocker/node/agents/test/test_blockdevice.py

```diff
@@ -49,6 +53,37 @@
     skip = "flocker.node.agents.blockdevice is only supported on Linux"
 
 
+def mkfs(device):
+    """
+    Synchronously initialize a device file with an ext4 filesystem.
+
+    :param FilePath device: The path to the file onto which to put the
+        filesystem.  Anything accepted by ``mkfs`` is acceptable (including a
+        regular file instead of a device file).
+    """
+    options = []
+    if not device.isBlockDevice():
+        options.extend([
+            # Make the filesystem even if the destination is not a block
+            # device.
+            b"-F",
```

Comment:
We ended up going via `/dev/loopN` everywhere else, I think.  That seems to be sufficient to avoid the need for `-F`.  I didn't want to do the extra work of setting up and tearing down loopback devices for the new tests I wrote where I just need some other random filesystem mounted in place of the correct volume.

Thinking about this now, maybe I should make this a flag instead of something automatically detected.  Then the new tests that definitely need it can pass the flag and the old tests won't have any chance of accidentally getting this behavior.